ONSdigital/dp-s3
# Migrated to latest version of healthcheck, and update code and tests

### What

- updated to newest version of health-check library
- removed Check object from Client (as now they are all centralised in the health-check library)
- added interface and mock for CheckState
- adapted the tests to new behaviour

### How to review

- Tests should pass
- Check that the code works
- If you want to validate it in a real system, do the following:
  - make sure you have a valid AWS credentials file (~/.aws/credentials), or exported env vars
  - modify the first unit test to provide a real s3client instead of the mock, i.e:
```
...
		// Create S3Client with mock AmzClient
		// s3AmzCliMock := &mock.AmzClientMock{
		// 	GetBucketReaderFunc: bucketExists,
		// }
		// s3Cli := s3client.NewWithClient(s3AmzCliMock, ExistingBucket)
		s3Cli, _ := s3client.New("eu-west-1", InexistentBucket)

		// mock CheckState for test validation
		mockCheckState := mock.CheckStateMock{
			UpdateFunc: func(status, message string, statusCode int) error {
				return nil
			},
		}
...
```
  - try to run with different combinations of regions and buckets (existing Bucket name exists in the real development system)

### Who can review

Anyone
